cmd/dmverity-vhd/main.go

```diff
@@ -0,0 +1,255 @@
+package main
+
+import (
+	"fmt"
+	"github.com/Microsoft/hcsshim/ext4/dmverity"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/pkg/errors"
+	"github.com/urfave/cli"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/Microsoft/hcsshim/ext4/tar2ext4"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+)
+
+const usage = `dmverity-vhd is a command line tool for creating LCOW layer VHDs with dm-verity hashes.`
+
+const (
+	usernameFlag  = "username"
+	passwordFlag  = "password"
+	imageFlag     = "image"
+	verboseFlag   = "verbose"
+	outputDirFlag = "out-dir"
+	maxVHDSize    = 128 * 1024 * 1024 * 1024
+)
+
+func main() {
+	cli.VersionFlag = cli.BoolFlag{
+		Name: "version",
+	}
+
+	app := cli.NewApp()
+	app.Name = "dmverity-vhd"
+	app.Commands = []cli.Command{
+		createVHDCommand,
+		rootHashVHDCommand,
+	}
+	app.Usage = usage
+	app.Flags = []cli.Flag{
+		cli.BoolFlag{
+			Name:  verboseFlag + ",v",
+			Usage: "Optional: verbose output",
+		},
+	}
+
+	if err := app.Run(os.Args); err != nil {
+		_, _ = fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+}
+
+func fetchImageLayers(ctx *cli.Context) (layers []v1.Layer, err error) {
+	image := ctx.String(imageFlag)
+	verbose := ctx.GlobalBool(verboseFlag)
+
+	ref, err := name.ParseReference(image)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to parse image reference: %s", image)
+	}
+
+	var remoteOpts []remote.Option
+	if ctx.IsSet(usernameFlag) && ctx.IsSet(passwordFlag) {
+		auth := authn.Basic{
+			Username: ctx.String(usernameFlag),
+			Password: ctx.String(passwordFlag),
+		}
+		authConf, err := auth.Authorization()
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to set remote")
+		}
+		if verbose {
```

Comment:
nit: maybe a wrapper func to get rid of the eight `if verbose`'s. Or I guess we could use logrus and set the log level to debug if verbose was specified and log these with `logrus.Debug`.